tools/hangar/__snapshots__/test_corpus/resource_captures_globals.w_compile_tf-aws.md

```diff
@@ -0,0 +1,500 @@
+# [resource_captures_globals.w](../../../../examples/tests/valid/resource_captures_globals.w) | compile | tf-aws
+
+## clients/Another.inflight.js
+```js
+class  Another {
+  constructor({ first, my_field, stateful }) {
+    this.first = first;
+    this.my_field = my_field;
+    this.stateful = stateful;
+  }
+  async $inflight_init()  {
+    {
+      {((cond) => {if (!cond) throw new Error(`assertion failed: '((await global_counter.peek()) === 0)'`)})(((await global_counter.peek()) === 0))};
+    }
+  }
+  async my_method()  {
+    {
+      (await global_counter.inc());
+      return (await global_counter.peek());
+    }
+  }
+}
+exports.Another = Another;
+let global_counter;
+exports.setupGlobals = function(globals) {
+  global_counter = globals["global_counter"];
+};
+
+```
+
+## clients/First.inflight.js
+```js
+class  First {
+  constructor({ my_resource, stateful }) {
+    this.my_resource = my_resource;
+    this.stateful = stateful;
+  }
+}
+exports.First = First;
+exports.setupGlobals = function(globals) {
+};
+
+```
+
+## clients/MyResource.inflight.js
+```js
+class  MyResource {
+  constructor({ stateful }) {
+    this.stateful = stateful;
+  }
+  async my_put()  {
+    {
+      (await global_bucket.put("key","value"));
+      {((cond) => {if (!cond) throw new Error(`assertion failed: '(global_str === "hello")'`)})((global_str === "hello"))};
+      {((cond) => {if (!cond) throw new Error(`assertion failed: '(global_bool === true)'`)})((global_bool === true))};
+      {((cond) => {if (!cond) throw new Error(`assertion failed: '(global_num === 42)'`)})((global_num === 42))};
+      {((cond) => {if (!cond) throw new Error(`assertion failed: '((await global_array_of_str.at(0)) === "hello")'`)})(((await global_array_of_str.at(0)) === "hello"))};
+      {((cond) => {if (!cond) throw new Error(`assertion failed: '((global_map_of_num)["a"] === (-5))'`)})(((global_map_of_num)["a"] === (-5)))};
+      {((cond) => {if (!cond) throw new Error(`assertion failed: '(await global_set_of_str.has("a"))'`)})((await global_set_of_str.has("a")))};
+      {((cond) => {if (!cond) throw new Error(`assertion failed: '(global_another.my_field === "hello!")'`)})((global_another.my_field === "hello!"))};
+      (await global_another.first.my_resource.put("key","value"));
+      {((cond) => {if (!cond) throw new Error(`assertion failed: '((await global_another.my_method()) > 0)'`)})(((await global_another.my_method()) > 0))};
+    }
+  }
+}
+exports.MyResource = MyResource;
```

Comment:
Perhaps I can suggest a different approach? (a bit more idiomatic to javascript):

Instead of a separate `setupGlobals` functions, why not just wrap the entire file in a function and use it to bind the global symbols:

```js
module.exports = function(globals) {
  class MyResource { ... /* simply references globals */ }
  return MyResource;
};
```

Then, the usage is:

```js
(await (async () => {
  const MyResource = require("${self_client_path}")({
    global_another: ${global_another_client},
    global_array_of_str: ${global_array_of_str_client},
    global_bool: ${global_bool_client},
    global_bucket: ${global_bucket_client},
    global_map_of_num: ${global_map_of_num_client},
    global_num: ${global_num_client},
    global_set_of_str: ${global_set_of_str_client},
    global_str: ${global_str_client},
  });
  const client = new MyResource({
    stateful: ${stateful_client},
  });
  if (client.$inflight_init) { await client.$inflight_init(); }
  return client;
})())
```

Reply:
Right on 👍